### PR TITLE
ci: always build images when core ci files change

### PR DESCRIPTION
**Pull request by nektro**

### CODE REVIEW

**Comment by robobun** [comment]
@nektro, your commit 8c6e7ac1c4c09f0cbf280029f208a8873ca568e5 is building: [`#6330`](https://buildkite.com/bun/bun/builds/6330)

File: .buildkite/ci.mjs [hunk 1]
```diff
@@ -664,12 +664,15 @@ async function main() {
...
     console.log("Checking changed files...");
-    const baseRef = getCommit();
+    const baseRef = lastBuild?.commit_id || getTargetBranch() || getMainBranch();
     console.log(" - Base Ref:", baseRef);
-    const headRef = lastBuild?.commit_id || getTargetBranch() || getMainBranch();
+    const headRef = getCommit();
     console.log(" - Head Ref:", headRef);
 
     changedFiles = await getChangedFiles(undefined, baseRef, headRef);
+    for (const f of changedFiles) {
```
Comment (Electroid) [inline .buildkite/ci.mjs]: `changedFiles` can be undefined intentionally, also the files are already printed out a few lines down

File: .buildkite/ci.mjs [hunk 2]
```diff
@@ -684,6 +687,7 @@ async function main() {
 
   console.log("Checking if CI should be forced...");
   let forceBuild;
+  let ciFileChanged;
```
Comment (Electroid) [inline .buildkite/ci.mjs:687]: Define this above the block that uses it

File: .buildkite/ci.mjs [hunk 3]
```diff
@@ -664,12 +664,15 @@ async function main() {
...
+    const baseRef = lastBuild?.commit_id || getTargetBranch() || getMainBranch();
     console.log(" - Base Ref:", baseRef);
-    const headRef = lastBuild?.commit_id || getTargetBranch() || getMainBranch();
+    const headRef = getCommit();
     console.log(" - Head Ref:", headRef);
 
     changedFiles = await getChangedFiles(undefined, baseRef, headRef);
+    for (const f of changedFiles) {
+      console.log(` - file: ${f}`);
+    }
```
Comment (nektro) [inline .buildkite/ci.mjs]:
```suggestion
```

File: .buildkite/ci.mjs [hunk 4]
```diff
@@ -693,6 +697,15 @@ async function main() {
       forceBuild = true;
     }
   }
+  {
```
Comment (nektro) [inline .buildkite/ci.mjs]:
```suggestion
```

**Pull request merged**